proto/viam/app/v1/app.proto

```diff
@@ -975,3 +979,23 @@ message CreateKeyResponse {
   string key = 1;
   string id = 2;
 }
+
+message DeleteKeyRequest {
+  string key = 1;
+  string id = 2;
+}
+
+message DeleteKeyResponse {}
+
+message ApiKey {
```

Comment:
@jr22 am going to copy and paste this into my current API Pr as we both seem to need it. Shouldn't be an issue to resolve for whoever gets there first :)

Reply:
also seeing EPs comment on your PR, reminded me that I actually probably need to include some more metadata here as well for the orgs page (authorizations etc)

Reply:
So on the frontend we would do something like: call `ListKeys` immediately followed by call to `ListAuthorizations` to get the details? Is that what you mean? 

If I'm understanding the code correctly, `ListOrganizationMembers` seems like it is querying the organization collection (mostly, although i do see the $lookups) and then `ListAuthorizations` queries the `identity authorizations` collection. In this case, I think its slightly different since we would be querying the `main.identity_authorization` collection for _both_ endpoints and for `ListKeys` we actually already have all the authorization information included in the db response, the current code is just filtering it out (due to slight misunderstanding by me for how this was going to be used). Given that it would be a duplicate db call, I _think_ it makes the most sense to include all the available info in the `ListKeys` response, but def open to push back since I'm still learning about the system!

Reply:
ah that's a fair point! also we probably will never need to return a key without its authorizations

Reply:
continuing convo from roxys pr here. updated mine to have one field that leverages APIKey and one that lists the authorizations. i'm not sooo in love with the key details being a separate field on the object from authorizations (feels like it implies that the authorizations aren't part of the key.. which they are) but also feel like this is the only way to get it to leverage the APIKey and avoid duplication. curious to hear what you guys think.

Reply:
i think it's a fine split because there is a valid use case for why you'd want just the API key and not its authorizations (ex. code samples). i feel like a user is pretty analogous in this situation, where it feels pretty natural that a user exists independently of its authorizations.

Reply:
I see what you mean here about the nesting, but yeah there is an argument that it also actually matches the DB object as that is nested. I would argue duplicating a few fields isn't that big of a deal,so I would favor just repeating the API key fields here and then adding the nested object for authorizations.